### ISSUE TITLE
Wrong classification model output 

### ISSUE THREAD

**Issue by AIGirl10** [Tencent/ncnn#2822]
Converted model from keras to NCNN. Pre-processing used in keras(python) is as follows

```py
# reading image in cv Mat format
orig = cv2.imread('black.png')
# Converting image from BGR to RGB
image = cv2.cvtColor(orig, cv2.COLOR_BGR2RGB)
# Resizing image to 128x128
image = cv2.resize(image, (128, 128))
# Convert image to float by and normalize by dividing it by 255
image = image.astype("float") / 255.0
# Convert image to array
image = img_to_array(image)
# Expand numpy array dimentions
image = np.expand_dims(image, axis=0)
# Load keras model
model = tf.keras.models.load_model('cnn128.h5')
# Give input to keras model
pred = model.predict(image)[0]
```

I tried to replicate the steps in c++ for NCNN model
```cpp
// reading image in cv Mat format
Mat image = imread("eyes.jpg"); //read image
// Converting image from BGR to RGB
cv::cvtColor(image, image, cv::COLOR_BGR2RGB);
// Resizing image to 128x128
Mat img;   
resize(image, img, Size(128, 128));
// Convert image to float by and normalize by dividing it by 255
img.convertTo(img,CV_32F);
img /= 255.0;
// Load ncnn model
ncnn::Net smallCNN;
smallCNN.load_param("cnn128.param");
smallCNN.load_model("cnn128.bin");
// converting cv::Mat image to ncnn::Mat
ncnn::Mat in = ncnn::Mat::from_pixels(img.data, ncnn::Mat::PIXEL_RGB, img.cols, img.rows);
// Give input to ncnn model
ncnn::Extractor ex = smallCNN.create_extractor();
ex.input("conv2d_8_input_blob", in);
```

when run the code, as an output I am getting this value

```sh
8 = 1.000000
7 = 0.000000
6 = 0.000000
```

where 8 is the label and 1.000000 is the probability. correct label is 0 and not 8

can anyone please help to understand what went wrong in input processing ?

**Comment by AIGirl10** [comment]
@nihui 

can you please check this ?

**Comment by superbayes** [comment]
// cv32fc3 的布局是 hwc ncnn的Mat布局是 chw 需要调整排布
float *srcdata = (float*)img.data;
float *data = new float[INPUT_WIDTH*INPUT_HEIGHT*3];
for (int i = 0; i < INPUT_HEIGHT; i++)
   for (int j = 0; j < INPUT_WIDTH; j++)
       for (int k = 0; k < 3; k++) {
          data[k*INPUT_HEIGHT*INPUT_WIDTH + i*INPUT_WIDTH + j] = srcdata[i*INPUT_WIDTH*3 + j*3 + k]/255.0;
       }
ncnn::Mat in(image.rows*image.cols*3, data);
in = in.reshape(image.rows, image.cols, 3);

// Load ncnn model
ncnn::Net smallCNN;
smallCNN.load_param("cnn128.param");
smallCNN.load_model("cnn128.bin");

[https://zhuanlan.zhihu.com/p/162542992](url)
